libraries/SPI/src/SPI.cpp

```diff
@@ -183,16 +183,20 @@ void SPIClassRP2040::beginTransaction(SPISettings settings) {
     DEBUGSPI("SPI::beginTransaction(clk=%lu, bo=%s)\n", settings.getClockFreq(), (settings.getBitOrder() == MSBFIRST) ? "MSB" : "LSB");
     if (_initted && settings == _spis) {
         DEBUGSPI("SPI: Reusing existing initted SPI\n");
-    } else {
-        _spis = settings;
-        if (_initted) {
-            DEBUGSPI("SPI: deinitting currently active SPI\n");
-            spi_deinit(_spi);
+    }
+    else {
+        /* Only de-init if the clock changes frequency */
+        if (settings.getClockFreq() != _spis.getClockFreq()) {
+          if (_initted) {
+              DEBUGSPI("SPI: deinitting currently active SPI\n");
+              spi_deinit(_spi);
+          }
+          DEBUGSPI("SPI: initting SPI\n");
+          spi_init(_spi, _spis.getClockFreq());
```

Comment:
This is backwards, actually.  `settings` is the new value (you've moved the copy `_spis = settings` after this block) so you want `spi_init(_spi, settings.getClockFreq());` I believe.